tests/build/build.go

```diff
@@ -516,6 +517,250 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
+
+				Expect(f.AsKubeAdmin.HasController.WaitForComponentPipelineToBeFinished(component, "", 2))
+				Expect(f.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(testNamespace))
+			})
+		})
+
+		Context("using PaC", func() {
+			BeforeAll(func() {
+				if utils.IsPrivateHostname(consoleRoute.Spec.Host) {
+					Skip("Using private cluster (not reachable from Github), skipping...")
+				}
+			})
+
+			It("configures PaC via annotation", func() {
+				Expect(f.AsKubeAdmin.HasController.SetComponentAnnotation(componentName, controllers.BuildRequestAnnotationName, controllers.BuildRequestConfigurePaCAnnotationValue, testNamespace)).To(Succeed())
+
+				var buildStatus *controllers.BuildStatus
+
+				Eventually(func() (bool, error) {
+					component, err := f.AsKubeAdmin.HasController.GetComponent(componentName, testNamespace)
+					if err != nil {
+						GinkgoWriter.Printf("cannot get the component: %v\n", err)
+						return false, err
+					}
+
+					statusBytes := []byte(component.Annotations[controllers.BuildStatusAnnotationName])
+
+					err = json.Unmarshal(statusBytes, &buildStatus)
+					if err != nil {
+						GinkgoWriter.Printf("cannot unmarshal build status: %v\n", err)
+						return false, err
+					}
+
+					if buildStatus.PaC != nil {
+						GinkgoWriter.Printf("state: %s\n", buildStatus.PaC.State)
+						GinkgoWriter.Printf("mergeUrl: %s\n", buildStatus.PaC.MergeUrl)
+						GinkgoWriter.Printf("errId: %d\n", buildStatus.PaC.ErrId)
+						GinkgoWriter.Printf("errMessage: %d\n", buildStatus.PaC.ErrMessage)
+						GinkgoWriter.Printf("configurationTime: %d\n", buildStatus.PaC.ConfigurationTime)
+					} else {
+						GinkgoWriter.Println("build status does not have PaC field")
+					}
+
+					return buildStatus.PaC != nil && buildStatus.PaC.State == "enabled" && buildStatus.PaC.MergeUrl != "" && buildStatus.PaC.ErrId == 0 && buildStatus.PaC.ConfigurationTime != "", nil
+				}, timeout, interval).Should(BeTrue(), "build status has unexpected content")
+			})
+
+			It("ensures build pipeline is triggered after merge", func() {
+				var pipelinerunPullSha string
+				var mergeResult *github.PullRequestMergeResult
+				var prNumber int
+
+				By("finding init PaC PR")
+				Eventually(func() (bool, error) {
+					prs, err := f.AsKubeAdmin.CommonController.Github.ListPullRequests(helloWorldComponentGitSourceRepoName)
+					if err != nil {
+						GinkgoWriter.Printf("cannot list pull requests: %v\n", err)
+						return false, err
+					}
+
+					for _, pr := range prs {
+						if pr.Head.GetRef() == branchName {
+							prNumber = pr.GetNumber()
+							pipelinerunPullSha = pr.Head.GetSHA()
+							return true, nil
+						}
+					}
+					return false, nil
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for init PaC PR to be created")
+
+				By("merging init PaC PR")
+				mergeResult, err := f.AsKubeAdmin.CommonController.Github.MergePullRequest(helloWorldComponentGitSourceRepoName, prNumber)
```

Comment:
Should we check for pull request pipelinerun before merging the pull request? Tough it might work perfectly well, I think we should follow the steps generally a user takes.